shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -81,6 +84,9 @@
     private final BroadcastReceiver mDiscoveryReceiver;
     private final List<ActivityLifecycleListener> mActivityLifecycleListeners;
     private long mNativePlatformView;
+    private boolean mIsUsingSoftwareRendering = false; // using the software renderer or not
+
+    private Bitmap mSoftwareRenderingBitmap;
```

Comment:
Declare this as volatile, and reassign it after copyPixelsFromBuffer() has completed on the new bitmap.  This will ensure that the writes performed on the GPU thread are visible to the UI thread when it reads mSoftwareRenderingBItmap.